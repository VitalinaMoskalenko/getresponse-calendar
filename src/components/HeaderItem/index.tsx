import React from "react";
import styled from "styled-components";
import { H3 } from "../Headings";

type HeaderPropsType = {
  isSelected: boolean;
};

const Container = styled.div<HeaderPropsType>`
  width: fit-content;
  padding: 10px;
  border-bottom: ${({ isSelected, theme }) =>
    isSelected ? `1px solid ${theme.colors.orange}` : "0px"};
  color: ${({ isSelected, theme }) =>
    isSelected ? `${theme.colors.orange}` : `${theme.colors.darkGrey}`};
`;

type PropsType = {
  titleHederItem: string;
  selectedHederItem?: boolean;
};

const HeaderItem = ({ selectedHederItem, titleHederItem }: PropsType) => {
  return (
    <Container isSelected={!!selectedHederItem}>
      <H3>{titleHederItem}</H3>
    </Container>
  );
};

export default HeaderItem;
