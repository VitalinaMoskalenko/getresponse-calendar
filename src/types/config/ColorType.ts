export enum ColorType {
  white = "white",
  black = "black",
  lightGrey = "#e1e1e1",
  darkGrey = "#535353",
  green = "#b2c200",
  orange = "#ff801a",
}
