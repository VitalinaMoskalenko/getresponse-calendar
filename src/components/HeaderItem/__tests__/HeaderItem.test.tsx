import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import HeaderItem from "../";
import { theme } from "../../../config";

it("renders HeaderItem correctly", () => {
  const headerItemComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <HeaderItem titleHederItem="" selectedHederItem />
      </ThemeProvider>
    )
    .toJSON();

  expect(headerItemComponent).toMatchSnapshot();
});
