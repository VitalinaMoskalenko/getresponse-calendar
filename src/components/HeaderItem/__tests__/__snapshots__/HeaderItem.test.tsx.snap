// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeaderItem correctly 1`] = `
<div
  className="sc-jSFjdj IZTbz"
>
  <h1
    className="sc-dlnjwi iBPsik"
  >
    
  </h1>
</div>
`;
